src/flow-rata-run-workspace/flow-rata-run-checks.service.ts

```diff
@@ -102,6 +124,82 @@ export class FlowRataRunChecksService {
     return errorList;
   }
 
+  private async rata116Check(
+    flowRataRun: FlowRataRunBaseDTO | FlowRataRunImportDTO,
+    referenceMethodCode: string,
+  ): Promise<string> {
+    let error: string = null;
+
+    // RATA-116-A: If CalculatedWAF is not null and ReferenceMethodCode is 2F, 2G, 2FJ, or 2GJ
+    if (flowRataRun.calculatedWAF !== null && flowRataRun.calculatedWAF !== undefined) {
+      if (['2F', '2G', '2FJ', '2GJ'].includes(referenceMethodCode)) {
+        error = CheckCatalogService.formatResultMessage('RATA-116-A', {
+          fieldname: 'calculatedWAF',
+          key: KEY
+        });
+      }
+
+      // RATA-116-B: If CalculatedWAF is less than 0 or greater than 1
+      if (flowRataRun.calculatedWAF < 0 || flowRataRun.calculatedWAF > 1) {
+        error = CheckCatalogService.formatResultMessage('RATA-116-B', {
+          fieldname: 'calculatedWAF',
+          key: KEY
+        });
+      }
+    } else {
+      // RATA-116-C: If CalculatedWAF is null and (ReferenceMethodCode is M2H or AverageVelocityWithWallEffects is not null)
+      if (referenceMethodCode === 'M2H' ||
+        flowRataRun.averageVelocityWithWallEffects !== null) {
+        error = CheckCatalogService.formatResultMessage('RATA-116-C', {
+          fieldname: 'calculatedWAF',
+          key: KEY
+        });
+      }
+    }
+
+    return error;
+  }
+
+  private async rata109Check(
+    rataRun: RataRun | RataRunImportDTO,
+    rataSummary: RataSummary | RataSummaryImportDTO,
+    testSumId?: string,
+  ): Promise<string> {
+    let error: string = null;
+
+    if (!rataSummary?.operatingLevelCode || rataRun.runNumber === null || rataRun.runNumber === undefined) {
+      return null;
+    }
+
+    let duplicateFound = false;
+
+    if (testSumId) {
+      const currentRataSummary = await this.rataSummaryRepository.findOne({
+        where: { id: (rataSummary as RataSummary).id },
+        relations: ['RataRuns']
```

Comment:
This works for now, but this method of specifying relations is deprecated and will be removed in the future.
```suggestion
        relations: {
          RataRuns: true,
        },
```